docs/.vitepress/theme/components/DiscordModal.vue

```diff
@@ -0,0 +1,70 @@
+<script setup lang="ts">
+import { onBeforeUnmount, onMounted, ref } from 'vue'
+
+const ONE_DAY_MS = 1000 * 60 * 60 * 24
+const STORAGE_KEY = 'discord-modal-last-opened'
+
+const isVisible = ref(false)
+
+const close = () => {
+  isVisible.value  = false
+  localStorage.setItem(STORAGE_KEY, String(new Date().getTime()))
+}
+
+onMounted(() => {
+  // get the last time we showed the modal, or 1 day ago if it's not set yet
+  const lastOpenedAt = localStorage.getItem(STORAGE_KEY) || (new Date().getTime()) - ONE_DAY_MS
+  const lastOpenDiff = new Date().getTime() - new Date(Number(lastOpenedAt)).getTime()
+
+  // show the newsletter modal once a day only
+  if (lastOpenDiff >= ONE_DAY_MS) {
+    // Show the modal 5 seconds after the page load
+    setTimeout(() => {
+      isVisible.value = true
+    }, 5000)
+  }
+})
+
+onBeforeUnmount(() => {
+  close()
```

Comment:
Should this check if the moda is visible?

Reply:
we can, but there's no actual benefit for that because this component only renders (yet it mounts) if the `isVisible` flag is true, (check line 35) so even if we dont check it, setting it false will remove the element from the dom tree and its bindings. We can check it inside the close function too but will be like having two `if` conditions of the same `isVisible` flag and with the same result.